wf-galleon/WFLY-19021-Stability_In_Provisioning.adoc

```diff
@@ -0,0 +1,294 @@
+
+
+=== Stability Level
+// Choose the planned stability level for the proposed functionality
+* [ ] Experimental
+
+* [ ] Preview
+
+* [x] Community
+
+* [ ] default
+
+
+=== Dev Contacts
+
+* mailto:{brian.stansberry@redhat.com}[Brian Stansberry]
+* mailto:{kkhan@redhat.com}[{Kabir Khan}]
+* mailto:{email}[{author}]
+
+=== QE Contacts
+
+* N/A.
+
+=== Testing By
+// Put an x in the relevant field to indicate if testing will be done by Engineering or QE. 
+// Discuss with QE during the Kickoff state to decide this
+* [x] Engineering
+
+* [ ] QE
+
+=== Affected Projects or Components
+
+* https://github.com/wildfly/wildfly[WildFly]
+* https://github.com/wildfly/galleon-plugins[WildFly Galleon plugins]
+* https://github.com/wildfly/galleon[Galleon]
+* https://github.com/wildfly/wildfly-glow[WildFly Glow]
+* https://github.com/wildfly/wildfly-maven-plugin[WildFly Maven Plugin]
+* https://github.com/wildfly-extras/wildfly-jar-maven-plugin[WildFly Bootable JAR]
+* https://github.com/wildfly-extras/prospero[Prospero]
+
+=== Relevant Installation Types
+// Remove the x next to the relevant field if the feature in question is not relevant
+// to that kind of WildFly installation
+* [x] Traditional standalone server (unzipped or provisioned by Galleon)
+
+* [x] Managed domain
+
+* [x] OpenShift s2i
+
+* [x] Bootable jar
+
+== Requirements
+
+Stability becomes a Galleon notion that applies to feature-pack, feature/feature_param and package.
+There is a 1 to 1 mapping with WildFly stability.
+The stability levels are: `default` >`community` > `preview` > `experimental`
+
+Galleon feature-pack descriptor is extended with 2 new attributes:
+
+* `config-stability-level`: The stability to be used when provisioning configurations.
+* `package-stability-level`: The stability to be used when provisioning packages.
+
+These attributes are set when building Galleon feature-packs and used by Galleon runtime at provisioning time.
+
+=== Requirements at feature-pack build time
+ 
+==== feature-pack minimum stability level
+
+A minimum stability level can be set at feature-pack build time. If no stability level is specified, the `default` level is used.
+
+Stability level is set thanks to a new option of the WildFly Galleon Plugins Maven plugin: 
+`<minimum-stability-level>default|community|preview|experimental</minimum-stability-level>`
+
+Feature-specs and packages at a lower stability level than is specified will not be included in the feature pack.
+
+==== feature-pack provisioning stability level
+
+This level is used as the default stability level to be used when provisioning server configurations and packages.
+If no stability level is specified, the `default` level is used for both configs and packages stability levels. 
+It is set thanks to a new option of the WildFly Galleon Plugins Maven plugin: 
+`<stability-level>default|community|preview|experimental</stability-level>`.
```

Comment:
Perhaps add a note:

If the specified stability level is not enabled by the level specified by the 'minimum-stability-level' setting, the build of the feature pack will fail.

Reply:
Or add a single note after the L140-143 note that covers all the cases I note here and in the next couple sections.

Reply:
Ok, added a note.